util/text.py

```diff
@@ -5,28 +5,34 @@
 from six.moves import range
 from functools import reduce
 
-# Constants
-SPACE_TOKEN = '<space>'
-SPACE_INDEX = 0
-FIRST_INDEX = ord('a') - 1  # 0 is reserved to space
-
-def text_to_char_array(original):
+class Alphabet(object):
+    def __init__(self, config_file):
+        self._label_to_str = []
+        self._str_to_label = {}
+        self._size = 0
+        with open(config_file, 'r') as fin:
+            for line in fin:
+                if line[0] == '#':
+                    continue
+                self._label_to_str += line[:-1] # remove the line ending
+                self._str_to_label[line[:-1]] = self._size
+                self._size += 1
+
+    def string_from_label(self, label):
+        return self._label_to_str[label]
+
+    def label_from_string(self, string):
+        return self._str_to_label[string]
+
+    def size(self):
+        return self._size
+
+def text_to_char_array(original, alphabet):
     r"""
     Given a Python string ``original``, remove unsupported characters, map characters
     to integers and return a numpy array representing the processed string.
     """
-    # Create list of sentence's words w/spaces replaced by ''
-    result = original.replace(" '", "") # TODO: Deal with this properly
-    result = result.replace("'", "")    # TODO: Deal with this properly
-
-    # Tokenize into letters adding in SPACE_TOKEN where required
-    result = np.hstack([SPACE_TOKEN if xt == ' ' else xt for xt in result])
-
-    # Map characters into indicies
-    result = np.asarray([SPACE_INDEX if xt == SPACE_TOKEN else ord(xt) - FIRST_INDEX for xt in result])
-
-    # Add result to results
-    return result
+    return np.asarray([alphabet.label_from_string(c) for c in original])
```

Comment:
What about the case of an `data/alphabet.txt` of the form
```

a
aa
...
```
In other words on with at least one occurrence of two characters per line. This doesn't handle this case.

My suggestion is to simply say multi-characters per line are not supported in `data/alphabet.txt` then leave this code as it.

Reply:
I'll reword the comment in the file.